### PR TITLE
KCE-76 Complete work on lock free queue

### PR DESCRIPTION
**Pull request by ThanosAndr**
[KCE-76](https://klepsydra.atlassian.net/browse/KCE-76)

[KCE-76]: https://klepsydra.atlassian.net/browse/KCE-76?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pablogh-2000 — COMMENTED** [review]
Is this ready for review?

File: README.md [hunk 1]
```diff
@@ -1,2 +1,114 @@
...
+# Installation Instructions
+
+## System dependencies
+
+* Ubuntu 14.04 or above
+* Cmake 3.5.1 or above
+* gcc for C++11 5.4.0 or above.
+* Google Test (https://github.com/google/googletest)
+* Google Benchmark (https://github.com/google/benchmark)
+
```
Comment (mateu94) [inline README.md:17]: Add a new section "Klepsydra dependencies" with:

- [kpsr-sdk](https://github.com/klepsydra-technologies/kpsr-sdk)
- [kpsr-core](https://github.com/klepsydra-technologies/kpsr-core)

File: README.md [hunk 2]
```diff
@@ -1,2 +1,114 @@
...
+```bash
+cd kpsr-queue-benchmarks/build
+./bin/queue_benchmarks_<name_of_the_benchmark>
+```
+Or run all benchmarks one after the other with the following command:
+
+```bash
+cd kpsr-queue-benchmarks/build
+for benchmark in *; do ./$benchmark; done
```
Comment (mateu94) [inline README.md:43]: The command must be run inside the bin folder, so there are 2 options:

1. Change the first line to: **cd kpsr-queue-benchmarks/build/bin**
2. Or, change the second line to: **for benchmark in ./bin/*; do $benchmark; done**

File: README.md [hunk 3]
```diff
@@ -1,2 +1,114 @@
...
+```
+
+## Overview
+Each benchmark measures the cpu impact of *push* and *pop* method implementations of the three target classes. The three classes implement different queues: kpsr::mem::SafeQueue: A Klepsydra implemented thread-safe asynchronous blocking queue and moodycamel::ConcurrentQueue: Is a thread-safe data structure that allows multiple threads to access it simultaneously. It is a specialized type of queue that is designed to handle multiple concurrent producers-consumers scenarios.
```
Comment (mateu94) [inline README.md:47]:
```suggestion
Each benchmark measures the cpu impact of *push* and *pop* method implementations of the three target classes. The three classes implement two different queues: 
1. kpsr::mem::SafeQueue: a Klepsydra implemented thread-safe asynchronous blocking queue.
2. moodycamel::ConcurrentQueue: a thread-safe data structure that allows multiple threads to access it simultaneously. It is a specialized type of queue that is designed to handle multiple concurrent producers-consumers scenarios.

```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,114 @@
...
+
+## Method Definition
+The measured methods are defined as follows: *push*: inserts the given element to the end of the queue. *pop*: removes an element from the front of the queue.
```
Comment (mateu94) [inline README.md:50]:
```suggestion
The measured methods are defined as follows:
- *push*: inserts the given element to the end of the queue.
- *pop*: removes an element from the front of the queue.

```

**Review by mateu94 — APPROVED** [review]

**Pull request merged**